src/test/java/core/basesyntax/RegistrationTest.java

```diff
@@ -0,0 +1,200 @@
+    }
+
+    @Test
+    public void test_hashCode_equalUsers_equalHashCodes() {
+        User user1 = new User();
+        user1.setLogin("user1");
+        user1.setPassword("password");
+        user1.setAge(20);
+
+        User user2 = new User();
+        user2.setLogin("user1");
+        user2.setPassword("password");
+        user2.setAge(20);
+
+        assertEquals(user1.hashCode(), user2.hashCode());
+    }
+
+    @Test
+    public void test_hashCode_differentUsers_differentHashCodes() {
+        User user1 = new User();
+        user1.setLogin("user1");
+        user1.setPassword("password");
+        user1.setAge(20);
+
+        User user2 = new User();
+        user2.setLogin("user2");
+        user2.setPassword("password");
+        user2.setAge(25);
+
+        assertNotEquals(user1.hashCode(), user2.hashCode());
+    }
+
+    @Test
+    public void test_gettersAndSetters_workOk() {
+        User user = new User();
+        user.setLogin("user1");
+        user.setPassword("password");
+        user.setAge(20);
+        user.setId(1L);
+
+        assertEquals("user1", user.getLogin());
+        assertEquals("password", user.getPassword());
+        assertEquals(20, user.getAge());
+        assertEquals(1, user.getId());
+
+        user.setLogin("newUser");
+        user.setPassword("newPassword");
+        user.setAge(25);
+        user.setId(2L);
+
+        assertEquals("newUser", user.getLogin());
+        assertEquals("newPassword", user.getPassword());
+        assertEquals(25, user.getAge());
+        assertEquals(2, user.getId());
+    }
+
+    /* Tests in class RegistrationServiceImpl */
+
+    @Test
+    public void test_checking_user_OnExisting() {
+        User user1 = new User();
+        user1.setLogin("Recruit");
+        user1.setPassword("password");
+        user1.setAge(20);
+        registrationService.register(user1);
+
+        User user2 = new User();
+        user2.setLogin("Recruit");
+        user1.setPassword("KingBoss");
+        user1.setAge(31);
+
+        assertThrows(InvalidDataException.class, () -> registrationService.register(user2));
+    }
+
+    @Test
+    public void test_loginLength_less6_notOk() {
+        User user = new User();
+        user.setLogin("PON");
+        user.setPassword("pon");
+        user.setAge(20);
+
+        assertThrows(InvalidDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void test_passwordLength_less6_notOk() {
+        User user = new User();
+        user.setLogin("PON");
+        user.setPassword("pon");
+        user.setAge(20);
+
+        assertThrows(InvalidDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void test_age_less18_notOk() {
+        User user = new User();
+        user.setLogin("PON");
+        user.setPassword("pon");
+        user.setAge(9);
+
+        assertThrows(InvalidDataException.class, () -> registrationService.register(user));
+    }
```

Comment:
Please, add test cases according to the checklist and comment in RegistrationServiceImpl 
Test cases: 
password / login == null
valid user -> ok

Read [checklist](https://github.com/mate-academy/jv-registration-validation/blob/master/checklist.md) file for more information